### PR TITLE
feat(api-v2): Accept custom new value IRI when updating value

### PR DESCRIPTION
**Pull request by benjamingeer**
https://dasch.myjetbrains.com/youtrack/issue/DSP-543

- [x] Accept a custom new value IRI when updating a value.
- [x] Require custom value IRIs to have the correct format.
- [x] Require custom resource IRIs to have the correct format.
- [x] Add tests and clarify some existing tests.
- [x] Add docs.

### CODE REVIEW

**Review by SepidehAlassi — APPROVED** [review]
It is great, thanks for this.

File: webapi/src/test/scala/org/knora/webapi/e2e/v2/ValuesRouteV2E2ESpec.scala [hunk 1]
```diff
@@ -2109,6 +2108,114 @@ class ValuesRouteV2E2ESpec extends E2ESpec {
...
+                propertyIriInResult = propertyIri,
+                expectedValueIri = intValueForRsyncIri.get,
+                userEmail = anythingUserEmail
+            )
+
+            val intValueAsInt: Int = savedValue.requireInt(OntologyConstants.KnoraApiV2Complex.IntValueAsInt)
+            intValueAsInt should ===(intValue)
+        }
+
+        "not update an integer value with a custom new value version IRI that is the same as the current IRI" in {
```
[the remark this answers is not in the document]
Comment (benjamingeer) [reply webapi/src/test/scala/org/knora/webapi/e2e/v2/ValuesRouteV2E2ESpec.scala:2147]: Good idea.

File: webapi/src/main/scala/org/knora/webapi/messages/StringFormatter.scala [hunk 2]
```diff
@@ -3160,4 +3160,28 @@ class StringFormatter private(val maybeSettings: Option[KnoraSettingsImpl] = Non
...
+
+    /**
+     * Validates a custom value IRI, throwing [[BadRequestException]] if the IRI is not valid.
+     *
+     * @param customValueIri the custom value IRI to be validated.
+     * @param projectCode the project code of the containing resource.
+     * @param resourceID the ID of the containing resource.
+     * @return the validated IRI.
+     */
+    def validateCustomValueIri(customValueIri: SmartIri, projectCode: String, resourceID: String): SmartIri = {
```
[the remark this answers is not in the document]
Comment (benjamingeer) [reply webapi/src/main/scala/org/knora/webapi/messages/StringFormatter.scala:3172]: No problem!

**Comment by benjamingeer** [comment]
Thanks for the review!

**Pull request merged**